qcodes_contrib_drivers/drivers/Tektronix/Keithley_6430.py

```diff
@@ -0,0 +1,431 @@
+# Qcodes driver Keithley 6430 SMU
+#
+# Joni Ikonnen <joni.2.ikonen@aalto.fi>, 2019 (port to Qcodes framework)
+# Russell Lake <russell.lake@aalto.fi>, 2012
+# Based on driver for Keithley 2700 by:
+# Pieter de Groot <pieterdegroot@gmail.com>, 2008
+# Martijn Schaafsma <qtlab@mcschaafsma.nl>, 2008
+# Reinier Heeres <reinier@heeres.eu>, 2008
+#
+# Update december 2009:
+# Michiel Jol <jelle@michieljol.nl>
+
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.utils.validators import Ints, Numbers, Bool, Strings, Enum
+from qcodes.utils.helpers import create_on_off_val_mapping
+import logging
+import warnings
+from functools import partial
+from typing import Tuple
+
+on_off_vals = create_on_off_val_mapping(on_val=1, off_val=0)
+
+
```

Comment:
has there not been problems with taking snapshot of this instruemnt? if there are, then it might be worth looking to various `snapshot_*` flags that `Parameter` exposes and use them for some of the parameters.

Reply:
@mgunyho what about this question?

Reply:
I'm not sure I entirely understand the question. Is it related to the comment

```
# NOTE that using 1/0 integer values will also work implicitly due to `hash(True) == hash(1)`/`hash(False) == hash(0)`
```
in `create_on_off_val_mapping`? We use `snapshot_value=False` in some of the parameters to supprress some warnings.

Reply:
> Is it related to the comment

nope :)

> We use snapshot_value=False in some of the parameters to supprress some warnings.

what warnings are being suppressed?

it is important. If these warnings can be addressed by careful use of snapshot related flags (`snapshot_value` , `snapshot_get`, `snapshot_exclude`) and some customization of parameter's set/get methods, then - great - because this will result in a more completed an meaningful snapshot.

-----------

the reason i initially asked the question is that this instrument seemingly has a few modes where in one mode values of some parameters can not be "gotten", and in another mode this is true for some other parameters. Indeed, the simple solution is just to exclude all of those parameters from the snapshot - but then the benefit of the snapshot is lost. So, i'm just reminding of a concern here, but it's up to you how to address it in this driver.

Reply:
You were right, the snapshot works just fine with `snapshot_get=False` and retrives the cached value if there is one. We probably had excluded those to suppress warnings in earlier development stage...
Anyway, this was fixed in d0dbdea